### PR TITLE
Add updates to FAQ for multi class classification and multiple/non-binary sensitive features

### PR DESCRIPTION
**Pull request by romanlutz**
While doing this update I also noticed that there was no entry for preprocessing in the corresponding table that I was linking to, so I quickly added that. Obviously, we'll need to add a user guide at some point.

@MiroDudik may want to take a look since he suggested that I create this PR in the first place :-)

Signed-off-by: Roman Lutz <rolutz@microsoft.com>

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -124,10 +124,28 @@ Which ML libraries does Fairlearn support?
     `predict()` methods. Also, any classification or regression
     algorithm can be evaluated using our metrics.
 
+Does Fairlearn support multi-class classification?
```
Comment (riedgar-ms) [inline docs/faq.rst:127]: I think that there should be 'metrics' and 'mitigation' portions to this FAQ.

File: docs/faq.rst [hunk 2]
```diff
@@ -124,10 +124,28 @@ Which ML libraries does Fairlearn support?
...
+    Certain metrics will still work for multi-class classification, e.g.,
+    accuracy score. For others it is more complicated since they rely on
+    binary classes, e.g., true positive rate.
```
Comment (riedgar-ms) [inline docs/faq.rst:130]: How about

> The `MetricFrame` does not care about the nature of the ML problem; if the supplied metric function supports multiclass classification, then the `MetricFrame` can use it, producing a result for each of the groups identified by the sensitive feature(s). For example, it is perfectly acceptable to pass `sklearn.metrics.confusion_matrix` as the metric function, and supply multiclass data for `y_true` and `y_pred`.

File: docs/faq.rst [hunk 3]
```diff
@@ -124,10 +124,28 @@ Which ML libraries does Fairlearn support?
...
+    There are
+    `ongoing discussions within the community <https://github.com/fairlearn/fairlearn/issues/752>`_
+    to add more extensive support to Fairlearn's assessment capabilities.
+    If you have specific thoughts feel free to add them to the discussion.
```
Comment (riedgar-ms) [inline docs/faq.rst:134]: Inviting 'specific thoughts' might make people concerned that their thoughts are insufficiently precise....

File: docs/faq.rst [hunk 4]
```diff
@@ -124,10 +124,28 @@ Which ML libraries does Fairlearn support?
...
+
+Does Fairlearn support multiple and non-binary sensitive features?
+    Fairlearn's assessment capabilities support sensitive features with more
+    than two values as well as multiple sensitive features.
+    Our :ref:`user guide <metrics_with_grouping>` has examples for both of
+    these cases.
+    The mitigation techniques all support mitigation with non-binary and
```
Comment (riedgar-ms) [inline docs/faq.rst:141]: While `GridSearch` can theoretically do this, I believe it runs into dimensional problems quite quickly.

Comment (romanlutz) [reply docs/faq.rst:141]: Yes, but I don't want that level of detail in this high-level FAQ. That belongs in the GridSearch API docs or user guide.

File: docs/faq.rst (hunk 2 above)
Comment (romanlutz) [reply docs/faq.rst:130]: I used this, but kept the part about adding thoughts since that is still evolving. Does that work for you?

File: docs/faq.rst (hunk 1 above)
Comment (romanlutz) [reply docs/faq.rst:127]: Indeed.

**Review by MiroDudik — APPROVED** [review]

File: docs/faq.rst [hunk 5]
```diff
@@ -124,10 +124,34 @@ Which ML libraries does Fairlearn support?
...
+    any metrics for supervised learning, including those for multi-class classification.
+    For example, it is possible to pass
+    :py:func:`sklearn.metrics.accuracy_score` or :py:func:`sklearn.metrics.confusion_matrix`
+    as the metric functions, and supply multi-class data for :code:`y_true` and :code:`y_pred`.
+    There are
+    `ongoing discussions within the community <https://github.com/fairlearn/fairlearn/issues/752>`_
+    to add more extensive support to Fairlearn's assessment capabilities.
+    If you have thoughts feel free to add them to the discussion.
+    
+    On the mitigation side, our algorithms :class:`ExponentiatedGradient` and :class:`GridSearch`
```
Comment (MiroDudik) [inline docs/faq.rst:138]: We have never tested this, but this should be allowed. I realize that our user guide doesn't say this... we can fix that another time.

**Review by riedgar-ms — APPROVED** [review]

**Comment by MiroDudik** [comment]
@romanlutz , @riedgar-ms : for some reason, I don't see the updated FAQ on CircleCI. any idea why?

**Comment by romanlutz** [comment]
> @romanlutz , @riedgar-ms : for some reason, I don't see the updated FAQ on CircleCI, so I'm hesitant to merge. any idea why?

I can't tell either. I re-queued it and will check whether it stays that way.

**Comment by romanlutz** [comment]
> > @romanlutz , @riedgar-ms : for some reason, I don't see the updated FAQ on CircleCI, so I'm hesitant to merge. any idea why?
> 
> I can't tell either. I re-queued it and will check whether it stays that way.

OK, (unsurprisingly) that didn't help. It does seem to check out the branch correctly, though. I just ran the single version sphinx command locally which rendered fine. @riedgar-ms @adrinjalali any ideas?

**Comment by MiroDudik** [comment]
[keeping fingers crossed!]

**Comment by romanlutz** [comment]
@MiroDudik thanks! It does show up now!

image
